src/Services/DBusServer.vala

```diff
@@ -0,0 +1,115 @@
+// -*- Mode: vala; indent-tabs-mode: nil; tab-width: 4 -*-
+/*-
+ * Copyright (c) 2017 elementary LLC. (https://elementary.io)
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ * Authored by: Adam Bieńkowski <donadigos159@gmail.com>
+ */
+
+[DBus (name="io.elementary.appcenter")]
+public class DBusServer : Object {
+    private static DBusServer? instance;
+    public static unowned DBusServer get_default () {
```

Comment:
Maybe construct your singleton using [GLib.Once](http://stackoverflow.com/questions/26832371/how-to-create-a-thread-safe-singleton-in-vala) for neatness?